midterm/shopping_list.py

```diff
@@ -0,0 +1,292 @@
+
+    def remove_from_food(self,item,quantity):
+        if item not in self.fridge_list:
+            print(f'The {item} is not in the fridge. Cannot take out from it.')
+        else:
+            self.fridge_list[item] -= quantity
+            if self.fridge_list[item] == 0:
+                del self.fridge_list[item]
+                print(f'You took out all the {item}. Go and shop some more.')
+            else:
+                print(f'We took out {quantity} piece(s) from the {item}.')
+
+
+    def __str__(self):
+        return super().__str__()
+
+    def check_recipe(self,recipe):
+
+        have_in_fridge = []
+        not_in_fridge = []
+        for item in recipe.ingredients.keys():
+            if item in self.fridge_list.keys():
+                have_in_fridge.append(item)
+            else:
+                not_in_fridge.append(item)
+                
+
+        print(f'Things for preparing {recipe.recipe_name} we don\'t have in the fridge: {not_in_fridge}')
+        print(f'Things for preparing {recipe.recipe_name} we have in the fridge: {have_in_fridge}')
+
+
+def check_the_fridge(fridge, recipe_list):
+
+    preparable_recipes = []
+    for name, ingredients in recipe_list.items():
+        count_recipe_ingredient = 0
+        for ingredient in ingredients.keys():
+            if ingredient in fridge:
+                count_recipe_ingredient += 1
+                if count_recipe_ingredient / len(recipe_list.values()) >= 0.5:
+                    preparable_recipes.append(name)
+
+    return f'Recipes we can make from the fridge: {preparable_recipes}'
+
+
+#shop={}
+def pretty_print_recipe(shop_display):
+    def wrapper1(*args):
+        shopping_list_arc=shop_display(*args)   #ramane
+        for key,value in shopping_list_arc:
+            #for k, v in key,value:
+            printer=f'{key}:{value}\n'
+        art_ascii = r"""
+   ______________________________
+ / \                             \.
+|   |                            |.
+ \_ |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |                            |.
+    |   _________________________|___
+    |  /                            /.
+    \_/dc__________________________/.
+            """
+        return printer+art_ascii
+    return wrapper1
+
+
+
+shopping_list_archive = []
+def archive_shopping_list(shop_ing):
+    def wrapper(*args):
+        shopping_list_archive.append(shop_ing(*args))
+        print (f'The {shop_ing(*args)} items were included in archive')
+        return shopping_list_archive 
+    return wrapper
+
+@archive_shopping_list
+#@pretty_print_recipe
```

Comment:
daca decoratorul archive_shoping_list returneaza `shop_ing(*args)` aici o sa puteti pune decoratorii in ordinea asta:
```
@pretty_print_recipe
@archive_shopping_list
def prepare_shopping_list()
```